tests/tests.rs

```diff
@@ -228,3 +228,202 @@ icp:public whatever
         )
         .success();
 }
+
+#[test]
+fn optimize() {
+    let expected_metadata = r#"icp:public candid:service
+icp:private candid:args
+icp:private motoko:stable-types
+icp:private motoko:compiler
```

Comment:
some of the canisters have a metadata section called "motoko"

Reply:
Maybe that's the name section? Check this

Reply:
`motoko` is generated by `moc` for upgrade compatibility check.

Reply:
So it should be preserved?

Reply:
Yes, everything that starts with "icp:" should be preserved.

Reply:
Oh sorry I meant this other custom section, that's not a icp metadata section.

For example, when I run `ic-wasm tests/motoko.wasm info`, the following is one of the outputs:

```
Custom sections with size: [
    "icp:public candid:service (77 bytes)",
    "icp:private candid:args (2 bytes)",
    "icp:private motoko:stable-types (56 bytes)",
    "icp:private motoko:compiler (6 bytes)",
    "motoko (15 bytes)",
]
```

Reply:
That might be the dwarf info? We only need to preserve the ones that starts with "icp:". Others are unreachable from the IC without that prefix.